zendesk/resource_zendesk_trigger_test.go

```diff
@@ -1,22 +1,199 @@
+			"active":      true,
+		},
+	}
+
+	trg, err := unmarshalTrigger(m)
+	if err != nil {
+		t.Fatalf("unmarshal returned an error: %v", err)
+	}
+
+	if v := m.Get("title"); trg.Title != v {
+		t.Fatalf("trigger had title value %v. should have been %v", trg.Title, v)
+	}
+}
+
+func TestCreateTrigger(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	m := mock.NewClient(ctrl)
+	i := newIdentifiableGetterSetter()
+	out := zendesk.Trigger{
+		ID:    12345,
+		Title: "trigger",
+	}
+
+	m.EXPECT().CreateTrigger(gomock.Any()).Return(out, nil)
+	if err := createTrigger(i, m); err != nil {
+		t.Fatal("CreateTrigger return an error")
+	}
+
+	if v := i.Id(); v != "12345" {
+		t.Fatalf("CreateTrigger did not set resource id. Id was %s", v)
+	}
+
+	if v := i.Get("title"); v != "trigger" {
+		t.Fatalf("CreateTrigger did not set resource title. title was %s", v)
+	}
+}
+
+func TestReadTrigger(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	m := mock.NewClient(ctrl)
+	i := newIdentifiableGetterSetter()
+	i.SetId("12345")
+
+	expected := zendesk.Trigger{
+		Title:  "trigger",
+		Active: true,
+	}
+	m.EXPECT().GetTrigger(gomock.Eq(int64(12345))).Return(expected, nil)
+	if err := readTrigger(i, m); err != nil {
+		t.Fatalf("GetTrigger received an error when calling: %v", err)
+	}
+}
+
+func TestUpdateTrigger(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	m := mock.NewClient(ctrl)
+	i := &identifiableMapGetterSetter{
+		id:              "12345",
+		mapGetterSetter: mapGetterSetter{},
+	}
+
+	m.EXPECT().UpdateTrigger(gomock.Eq(int64(12345)), gomock.Any()).Return(zendesk.Trigger{}, nil)
+	if err := updateTrigger(i, m); err != nil {
+		t.Fatalf("updateTrigger returned an error %v", err)
+	}
+}
+
+func TestDeleteTrigger(t *testing.T) {
 	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
 	c := mock.NewClient(ctrl)
 	d := newIdentifiableGetterSetter()
 
 	d.SetId("1234")
 
 	c.EXPECT().DeleteTrigger(gomock.Eq(int64(1234))).Return(nil)
-	err := resourceZendeskTriggerDelete(d, c)
+	err := deleteTrigger(d, c)
 	if err != nil {
 		t.Fatalf("Got error from resource delete: %v", err)
 	}
 }
+
+func testTriggerDestroyed(s *terraform.State) error {
+	client := testAccProvider.Meta().(zendesk.TriggerAPI)
+
+	for k, r := range s.RootModule().Resources {
+		if r.Type != "zendesk_trigger" {
+			continue
+		}
+
+		id, err := atoi64(r.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		_, err = client.GetTrigger(id)
+		if err == nil {
+			return fmt.Errorf("did not get error from zendesk when trying to fetch the destroyed trigger. resource name %s", k)
+		}
+
+		zdresp, ok := err.(zendesk.Error)
+		if !ok {
+			return fmt.Errorf("error %v cannot be asserted as a zendesk error", err)
+		}
+
+		if zdresp.Status() != http.StatusNotFound {
+			return fmt.Errorf("did not get a not found error after destroy. error was %v", zdresp)
+		}
+	}
+	return nil
+}
+
+func TestAccTriggerExample(t *testing.T) {
```

Comment:
When i ran this locally i got the following error.

```
--- FAIL: TestAccTriggerExample (0.26s)
    testing.go:538: Step 0 error: Error applying: 1 error occurred:
        	* zendesk_trigger.auto-reply-trigger: 1 error occurred:
        	* zendesk_trigger.auto-reply-trigger: 422: {"error":"RecordInvalid","description":"Record validation errors","details":{"base":[{"description":"Trigger must contain at least one action"}]}}
```

Reply:
Oh, I got same error.
It seems that the error is caused by difference of type bc only notification_user uses not string but []string.

Thank you for your report. I’ll try to fix it up.